### DISCUSSION TITLE
MarkerClusterer component?

### DISCUSSION THREAD

**Discussion by hydrohiester** [visgl/react-google-maps#644] [General]
Any chance there is a <MarkerClusterer /> component on the roadmap?  (I know there is are examples...I like the abstraction of a component). If nobody is building one, maybe I'll take a crack at it.

**Comment by usefulthink** [comment]
Generally yes, but I think it's going to be difficult to come up with a solution that is good enough to be useful and at the same time flexible enough to handle more than just a few narrow use-cases:

 - both individual- and cluster-markers should be customizable in a simple way. Ideally, users will provide components for the different types of markers (what's the best way to do this? Callback functions rendering the markers?)
 - it should be able to work with all sorts of different data sources. The narrower this definition gets, the more preprocessing needs to be done by users. But generic solutions that just work with whatever data the user has are tricky to implement. There's a sweet spot somewhere...

If you like to give it a try, that would be really cool. Maybe you could start with the [custom marker clustering example](https://visgl.github.io/react-google-maps/examples/custom-marker-clustering) example and wrap it into a component that allows customizing the markers and works with any array of objects as input.
